### PR TITLE
improve pnpm support, fixes xcode build fails

### PR DESCRIPTION
**Pull request by franksmule**
# Why

- When building from xcode ( After a expo prebuild, opening the xcode workspace in the Xcode app and building )  with pnpm in a monorepo - xcode cannot find `expo-modules-autolinking` and `@expo/cli`
- Other resolution issues can be fixed with metro.config.js - but these require patching

# How

Followed @byCedric example - https://github.com/expo/expo/pull/23937 and added it to two more places! 

The xcode project diff it hard to follow but it just changes;

`rrequire.resolve('@expo/cli')` to 
`require.resolve('@expo/cli', { paths: [require.resolve('expo/package.json')]})`

# Test Plan

Forked https://github.com/byCedric/expo-monorepo-example to
https://github.com/franksmule/expo-monorepo-example.

Uses patches + packed template.

Without patch & template do prebuild then open in xcode to build, see build errors below:

`Error: Cannot find module 'expo-modules-autolinking'`
`Error: Cannot find module '@expo/cli'`

Apply patch and use fixed template, build succeeds.

# Checklist
- [x] No documentation changes.
- [x] This diff will work correctly for `npx expo prebuild`
- [ ] EAS build not tested

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]
Looks good to me, thanks for contribution! 🙏

**Review by Kudo — APPROVED** [review]

**Review by byCedric — APPROVED** [review]

**Comment by franksmule** [comment]
Thanks for the reviews @byCedric @Kudo @tsapeta - the workflows won't working so I rebased - do you mind kicking off again?

**Comment by byCedric** [comment]
Can you add a changelog entry to https://github.com/expo/expo/blob/main/packages/expo-modules-autolinking/CHANGELOG.md under "Bugfixes"? You can copy one of the other entries :)

**Comment by byCedric** [comment]
PR #25817 is up, which should supersede this PR

I added you as co-author to the two commits 😄
